Drv/TcpClient/CMakeLists.txt

```diff
@@ -7,6 +7,12 @@
 ####
 restrict_platforms(Posix SOCKETS)
 
+if (BUILD_TESTING)
+    add_compile_options(
```

Comment:
Should we add the `-Wno-conversion` to just the test target rather than the whole module when `BUILD_TESTING`?

Reply:
This would look like:

```
register_fprime_ut()
if (BUILD_TESTING)
    target_compile_options("${FPRIME_CURRENT_MODULE}_ut_exe")
endif()
```

Reply:
Done.

Reply:
That formula caused CI to explode, because of differences in compilation on the workstation and in CI. Using the revised formula provided by @LeStarch.